website/content/docs/connect/cluster-peering/k8s.mdx

```diff
@@ -97,7 +97,41 @@ Install Consul on Kubernetes by using the CLI to apply `values.yaml` to each clu
 
 ## Create a peering connection for Consul on Kubernetes
 
-To peer Kubernetes clusters running Consul, you need to create a peering token and share it with the other cluster. Complete the following steps to create the peer connection.
+To peer Kubernetes clusters running Consul, you need to create a peering token on one cluster (`cluster-01`) and share
+it with the other cluster (`cluster-02`). The generated peering token from `cluster-01` will include the addresses of
+the servers for that cluster. The servers for `cluster-02` will use that information to dial the servers in
+`cluster-01`. Complete the following steps to create the peer connection.
+
+### Using mesh gateways for the peering connection
+If the servers in `cluster-01` are not directly routable from the dialing cluster `cluster-02`, then you'll need to set up peering through mesh gateways.
+
+1. In `cluster-01` apply the `Mesh` custom resource so the generated token will have the mesh gateway addresses which will be routable from the other cluster.
+
+    <CodeBlockConfig filename="mesh.yaml">
```

Comment:
I think Vercel might be upset because it needs closing tags for `CodeBlockConfig`

Reply:
Ahh thanks!!